cql3/statements/function_statement.cc

```diff
@@ -27,6 +28,45 @@ future<> create_function_statement_base::check_access(query_processor& qp, const
     }
 }
 
+future<> drop_function_statement_base::check_access(query_processor& qp, const service::client_state& state) const
+{
+    create_arg_types(qp);
+    shared_ptr<functions::function> func;
+    if (_args_present) {
+        func = functions::functions::find(_name, _arg_types);
+        if (!func) {
+            return make_ready_future<>();
+        }
+    } else {
+        auto funcs = functions::functions::find(_name);
+        if (!funcs.empty()) {
+            auto b = funcs.begin();
+            auto i = b;
+            if (++i != funcs.end()) {
+                return make_ready_future<>();
+            }
+            func = b->second;
+        } else {
+            return make_ready_future<>();
+        }
+    }
+
+    sstring encoded_signature = format("{}[{}]",
+            _name.name,
+            ::join("^", _arg_types | boost::adaptors::transformed([] (const data_type t) {
+                return t->name();
+            })));
+
+    try {
+        return state.has_function_access(qp.db(), _name.keyspace, encoded_signature, auth::permission::DROP);
+    } catch (exceptions::invalid_request_exception&) {
+        if (!_if_exists) {
+            throw;
```

Comment:
From what I see, at this point in code we're not checking whether the function exists any more (we did that above), and we're only working with the `resource` object.
Consequently, the `invalid_request_exception` is actually only thrown if the given keyspace is empty: https://github.com/scylladb/scylladb/blob/25cf3256743d98f1dd34ef0656ee7cc974f87249/service/client_state.cc#L105-L108 which makes no sense here because we've already confirmed that our function is in this keyspace.

I think I'll remove the throw anyway, because as you've mentioned, `check_access()` is for checking permissions, and  `invalid_request_exception` is not an authorization problem